src/core/core.layouts.js

```diff
@@ -48,10 +98,137 @@ function findMaxPadding(boxes) {
 	};
 }
 
-function addSizeByPosition(boxes, size) {
-	helpers.each(boxes, function(box) {
-		size[box.position] += box.isHorizontal() ? box.height : box.width;
-	});
+function addSizeByPosition(size, boxes) {
+	var result = extend({}, size);
+	var i, ilen, layout, box;
+	for (i = 0, ilen = boxes.length; i < ilen; ++i) {
+		layout = boxes[i];
+		box = layout.box;
+		result[box.position] += layout.isHorizontal ? box.height : box.width;
+	}
+	return result;
+}
+
+function calculateMinSizes(boxes, params) {
+	var chartArea = params.chartArea;
+	var i, ilen, layout, box, minSize;
+
+	for (i = 0, ilen = boxes.length; i < ilen; ++i) {
+		layout = boxes[i];
+		box = layout.box;
+
+		layout.isHorizontal = layout.box.isHorizontal();
+		if (layout.isHorizontal) {
+			minSize = box.update(box.fullWidth ? params.width : chartArea.w, params.hBoxMaxHeight);
+			chartArea.h -= minSize.height;
+		} else {
+			minSize = box.update(params.vBoxMaxWidth, chartArea.h);
+			chartArea.w -= minSize.width;
+		}
+		layout.minSize = minSize;
+	}
+	return addSizeByPosition(params.padding, boxes);
+}
+
+function fitHorizontalBoxes(boxes, params, size) {
+	var maxPadding = params.maxPadding;
+	var i, ilen, layout, box, margin;
+
+	for (i = 0, ilen = boxes.length; i < ilen; ++i) {
+		layout = boxes[i];
+		box = layout.box;
+
+		margin = {
+			left: Math.max(size.left, maxPadding.left),
+			right: Math.max(size.right, maxPadding.right),
+			top: 0,
+			bottom: 0
+		};
+
+		// Don't use min size here because of label rotation. When the labels are rotated, their rotation highly depends
+		// on the margin. Sometimes they need to increase in size slightly
+		box.update(box.fullWidth ? params.width : params.chartArea.w, params.hBoxMaxHeight, margin);
+	}
+	return addSizeByPosition(size, boxes);
+}
+
+function fitVerticalBoxes(boxes, params, size, finalFit) {
+	var maxPadding = params.maxPadding;
+	var i, ilen, layout, box, margin;
+
+	for (i = 0, ilen = boxes.length; i < ilen; ++i) {
+		layout = boxes[i];
+		box = layout.box;
+
+		margin = finalFit && {
+			left: 0,
+			right: 0,
+			top: Math.max(size.top, maxPadding.top),
+			bottom: Math.max(size.bottom, maxPadding.bottom)
+		};
+
+		box.update(params.vBoxMaxWidth, params.chartArea.h, margin);
+	}
+	return finalFit ? size : addSizeByPosition(size, boxes);
+}
+
+function updateChartArea(params, sizes, updateSizes) {
+	var maxPadding = params.maxPadding;
+	var padding = params.padding;
+	var chartArea = params.chartArea;
+	var oldWidth = chartArea.w;
+	var oldHeight = chartArea.h;
+	var left = Math.max(maxPadding.left, sizes.left);
+	var top = Math.max(maxPadding.top, sizes.top);
+	var right = Math.max(maxPadding.right, sizes.right);
+	var bottom = Math.max(maxPadding.bottom, sizes.bottom);
+	var leftAdd = left - sizes.left;
+	var topAdd = top - sizes.top;
+
+	// We may be adding some padding to account for rotated x axis labels
+	if (updateSizes) {
+		sizes.left = left;
```

Comment:
I would probably do this directly in `update` rather than adding an additional parameter to the method and making the method do two things

Reply:
Removed the parameter and just update those sizes always. Also renamed the function.